### PR TITLE
Task-36267 : when saml is configured, and when you save a wiki page c…

### PR DESCRIPTION
**Pull request by rdenarie**
…ontaining accentued char, there is an encoding problem

Before this fix, the SSODelegateValve checks if we are in logout process, and for that check a parameter in url. But at this point, the characterEncoding for this request is not set. It is set later.
This fix add the character encoding if not present.

### CODE REVIEW

File: saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java [hunk 1]
```diff
@@ -382,6 +382,13 @@ private boolean isGlobalLogout(Request request) {
      * @return true if this is a local SAML logout
      */
     private boolean isLocalLogout(Request request) {
+        try {
+            if (request.getCharacterEncoding()==null) {
+                request.setCharacterEncoding("UTF-8");
+            }
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
```
Comment (ahamdi) [inline saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java]: We need to use the logger instaed.

File: saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java [hunk 2]
```diff
@@ -382,6 +382,13 @@ private boolean isGlobalLogout(Request request) {
      * @return true if this is a local SAML logout
      */
     private boolean isLocalLogout(Request request) {
+        try {
+            if (request.getCharacterEncoding()==null) {
```
Comment (ahamdi) [inline saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java:386]: I am not sure we need to check as we have to force the encoding to UTF-8, can we just set its value to UTF-8 ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java (hunk 2 above)
Comment (rdenarie) [reply saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java:386]: In fact, if the request have already an encoding, I dont want to enforce it. It should keep the existing one

File: saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java (hunk 1 above)
Comment (rdenarie) [reply saml/gatein-saml-plugin/src/main/java/org/gatein/sso/saml/plugin/valve/AbstractSPFormAuthenticator.java]: done

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**